proposals/propertyDependencies.md

```diff
@@ -0,0 +1,292 @@
+# JSON Schema: The `propertyDependencies` Keyword
+
+## Abstract
+
+The `propertyDependencies` keyword is a more friendly way to select between two
+or more schemas to validate an instance against than is currently supported by
+JSON Schema.
+
+## Status
+
+**Current Status**: PROPOSAL
+
+TODO: We should have a short standard blurb outlining the stages involved in a
+feature making its way to stable status.
+
+TODO: Link to a document that describes the proposal => stable process in
+detail.
+
+## Note to Readers
+
+The issues list for this document can be found at
+<https://github.com/json-schema-org/json-schema-spec/issues?q=is%3Aissue+propertydependencies>.
+
+For additional information, see <https://json-schema.org/>.
+
+To provide feedback, use this issue tracker or any of the communication methods
+listed on the homepage.
+
+## Table of Contents
+
+## Conventions and Terminology
+
+All conventions and terms used and defined by the [JSON Schema Core
+specification](../jsonschema-core.html) also apply to this document.
+
+## Overview
+
+### Problem Statement
+A common need in JSON Schema is to select between one schema or another to
+validate an instance based on the value of some property in the JSON instance.
+There are a several patterns people use to accomplish this, but they all have
+significant [problems](#problems).
+
+OpenAPI solves this problem with the `discriminator` keyword. However, their
+approach is more oriented toward code generation concerns, is poorly specified
+when it comes to validation, and is couple to OpenAPI concepts that don't exist
+is JSON Schema. Therefore, it's necessary to define something new rather than
+adopt `discriminator`.
+
+### Solution
+
+The `dependentSchemas` keyword is very close to what is needed except it checks
+for the presence of a property rather than it's value. The chosen solution is to
+build on that concept to solve this problem.
+
+```jsonschema
+{
+  "propertyDependencies": {
+    "foo": {
+      "aaa": { "$ref": "#/$defs/foo-aaa" }
+    }
+  }
+}
+```
+
+The validation result is equivalent to the following schema.
+
+```jsonschema
+{
+  "if": {
+    "properties": {
+      "foo": { "const": "aaa" }
+    },
+    "required": ["foo"]
+  },
+  "then": { "$ref": "#/$defs/foo-aaa" }
+}
+```
+
+### Limitations
+
+The problem of choosing an alternative based on a property value could apply for
```

Comment:
Maybe choose a different word from "alternative" here since it has special meaning in the context of this document.

```suggestion
The problem of choosing a subschema based on a property value could apply for
```

Reply:
Good call. I didn't notice that.